cmd/sniproxy/sniproxy.go

```diff
@@ -83,12 +138,35 @@ func main() {
 		go s.promoteHTTPS(ln)
 	}
 
+	if *debugPort != 0 {
+		mux := http.NewServeMux()
+		tsweb.Debugger(mux)
+		srv := &http.Server{
+			Handler: mux,
+		}
+		dln, err := s.ts.Listen("tcp", fmt.Sprintf(":%d", *debugPort))
+		if err != nil {
+			log.Fatal(err)
+		}
+		go func() {
+			log.Fatal(srv.Serve(dln))
+		}()
+	}
+
 	select {}
 }
 
 type server struct {
 	ts tsnet.Server
 	lc *tailscale.LocalClient
+
+	stats          *metrics.Set
+	numTLSsessions *expvar.Int
```

Comment:
(optional) Since we pass the server around by pointer anyway, we could also store these by value and just take their address in `initMetrics` (then we don't need to initialize them explicitly)

Reply:
Done, except for the LabelMap which takes an argument in its initialization.